src/deploy/functions/runtimes/node/parseTriggers.ts

```diff
@@ -150,9 +152,13 @@ export function addResourcesToBackend(
     }
 
     if (annotation.httpsTrigger) {
-      trigger = {
-        allowInsecure: true,
-      };
+      let allowInsecure: boolean;
+      if ("allowInsecure" in annotation.httpsTrigger) {
+        allowInsecure = !!annotation.httpsTrigger.allowInsecure;
+      } else {
+        allowInsecure = annotation.apiVersion != 2;
```

Comment:
Looks like we need to change this to platform now?

Reply:
Yup. Changing.